Settings/PricePartSettings.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace OrchardCore.Commerce.Settings
+{
+    public enum CurrencySelectionModeEnum
+    {
+        AllCurrencies,
+        DefaultCurrency,
+        SpecificCurrency
+    }
+
+    public class PricePartSettings
+    {
+        public string CurrencySelectionMode { get; set; }
```

Comment:
Did you miss this one?

Reply:
I kept this as a string for serialization. That was the reason I made them strings to start with. I'll look at it again and see if I can solve this.

Reply:
Serialization of the enum should work fine. I don't think you need to keep this as string.

Reply:
It's changed to be an enum everywhere now. It's serialized as numeric. I don't get the string conversion to work. It work, but is serialized as a number. Have to look at how the Json is serialized in Orchard Core.

Reply:
If you want the string representation, use the `StringEnumConverter`. `[JsonConverter(typeof(StringEnumConverter))]` on the property.

Reply:
I tried that both with the Newtonsoft and the new System.Text.Json attributes for string convertion. But I didn't get it to work. It's ok to store it as the enum number for now.

Reply:
Where did you put the attribute? (also FWIW, I'll probably remove all the System.Text.Json converters, they were a failed experiment)

Reply:
I put it on the enum property in PricePartSettings.

Reply:
It works now. [JsonObject] on the class was missing. Have made a new push with this change.